From 63a4b327ab23b55ff886ddb257beaade5da0b7c5 Mon Sep 17 00:00:00 2001
From: Xulin Sun <xulin.sun@windriver.com>
Date: Sun, 27 Sep 2020 14:35:25 +0800
Subject: [PATCH] pfeng: add GCC version 10.2.0 support

Add GCC version 10.2.0 support, to fix below compile error:
../drivers/net/pfeng/hw_abi/pfe_compiler.h:90:2: error: #error Please specify your compiler behavior.
   90 | #error Please specify your compiler behavior.
      |  ^~~~~
../drivers/net/pfeng/hw_abi/pfe_compiler.h:96:2: error: #error Invalid value of PFE_COMPILER_BITFIELD_BEHAVIOR
   96 | #error Invalid value of PFE_COMPILER_BITFIELD_BEHAVIOR
      |  ^~~~~
In file included from ../drivers/net/pfeng/hw_abi/pfeng_hw.h:42,
                 from ../drivers/net/pfeng/pfeng.h:11,
                 from ../drivers/net/pfeng/pfeng_cmd.c:26:

Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/net/pfeng/hw_abi/pfe_compiler.h | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/drivers/net/pfeng/hw_abi/pfe_compiler.h b/drivers/net/pfeng/hw_abi/pfe_compiler.h
index 1e0a064e51..1e925f6e1e 100644
--- a/drivers/net/pfeng/hw_abi/pfe_compiler.h
+++ b/drivers/net/pfeng/hw_abi/pfe_compiler.h
@@ -63,6 +63,17 @@
 #if (defined(PFE_CFG_TARGET_ARCH_aarch64))
 #define PFE_COMPILER_BITFIELD_BEHAVIOR PFE_COMPILER_BITFIELD_HIGH_LAST
 #endif
+#elif ((__GNUC__ == 10) && (__GNUC_MINOR__ == 2) && (__GNUC_PATCHLEVEL__ == 0))
+/* GCC version 10.2.0 */
+#if (defined(PFE_CFG_TARGET_ARCH_aarch64))
+/* Compiling driver */
+#define PFE_COMPILER_BITFIELD_BEHAVIOR PFE_COMPILER_BITFIELD_HIGH_LAST
+#endif
+#elif defined(PFE_CFG_TARGET_OS_UBOOT)
+/* U-boot wants NO compiler check, so pass down */
+#if (defined(PFE_CFG_TARGET_ARCH_aarch64))
+#define PFE_COMPILER_BITFIELD_BEHAVIOR PFE_COMPILER_BITFIELD_HIGH_LAST
+#endif
 #endif
 #endif
 
-- 
2.17.1

